docs/design_proposals/user_survey_hooks.md

```diff
@@ -0,0 +1,135 @@
+# Title
+
+* Author(s): Tejal Desai
+* Design Shepherd: Brian de Alwis
+* Date: 07/11/2021
+* Status: Proposed
+
+## Background
+
+Currently, we get feedback on skaffold via skaffold HaTs survey. 
+However, in this survey we don't get feedback on new or existing features. 
+We also can't get enough information from our existing customers on how they are using certain features in case we want to make any changes.
+e.g.
+With render v2, we are changing how deploy works. This may affect our existing helm deployer users. 
+See [#6166](https://github.com/GoogleContainerTools/skaffold/issues/6166)
+## Design
+
+### User survey Definition
+In order to make sure only relevant surveys are shown to user, we have added a survey config.
+```
+type config struct {
+	id           string
+	promptText   string
```

Comment:
Worth describing what is expected here.  Is this the only text to be shown, or will we add other boilerplate?
```suggestion
	// promptText is shown to the user and should be formatted so each line should fit in < 80 characters.
	// For example: `As a Helm user, we are requesting your feedback on a proposed change to Skaffold's integration with Helm.`
	promptText   string
```